frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -43,6 +46,66 @@ export const insightLogic = kea<insightLogicType>({
                 setAllFilters: (_, { filters }) => filters,
             },
         ],
+        /*
+        isFirstLoad determines if this is the first graph being shown after the component is mounted (used for analytics)
+        */
+        isFirstLoad: [
+            true,
+            {
+                setNotFirstLoad: () => false,
+            },
+        ],
+    }),
+    listeners: ({ values, actions }) => ({
+        setAllFilters: (filters) => {
+            actions.reportUsage(filters.filters)
+            actions.setNotFirstLoad()
+        },
+        reportUsage: async (filters: Record<string, any>, breakpoint) => {
```

Comment:
What's the real type for filters here? Since it only seems to have 4 valid keys, can we create a proper interface for it?

Without doing so, all of the keys below will have type `any` and we're basically not making use of typescript.

We can't really make use of exhaustive pattern matching for `insight` type as well - it's possible to write this code in a way that if a new view is added, this place will start to error and we'll know to add another case to the `if` below.

Happy to pair on this.

Reply:
I think it's a bit trickier, filters can really be string, number, an object, an array of strings/numbers/objects. They also seem to have a ton more keys. 100% agree we should type this better, but might be outside the scope of this? Will ping you to see if it's worth fixing rn.